yt_dlp/extractor/wikimedia.py

```diff
@@ -0,0 +1,69 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+from .common import InfoExtractor
+
+import re
+
+from ..utils import (
+    clean_html,
+    determine_ext,
+    get_element_by_class,
+    urljoin,
+    parse_qs as compat_parse_qs,
+    ExtractorError)
+
+
+class WikimediaIE(InfoExtractor):
+    IE_NAME = 'wikimedia.org'
+    _NETRC_MACHINE = 'wikimediaorg'
+    _API_BASE_URL = 'https://commons.wikimedia.org/'
+    _VALID_URL = 'https://commons.wikimedia.org/wiki/File:(?P<id>[^/]+)'
+
+    _TEST = {
+        'url': 'https://commons.wikimedia.org/wiki/File:Die_Temperaturkurve_der_Erde_(ZDF,_Terra_X)_720p_HD_50FPS.webm',
+        'info_dict': {
+            'description': 'md5:7cd84f76e7081f1be033d0b155b4a460',
+            'ext': 'webm', 'id': 'Die_Temperaturkurve_der_Erde_(ZDF,_Terra_X)_720p_HD_50FPS',
+            'title': 'Die Temperaturkurve der Erde (ZDF, Terra X) 720p HD 50FPS.webm - Wikimedia Commons',
+            'license': 'md5:62907cddf705a9f7ae7076c15407a977',
+            'author': None, 'subtitles': {'de': [{'ext': 'srt',
+                                                  'url': 'https?://commons.wikimedia.org/w/api.php'}],
+                                          'en-gb': [{'ext': 'srt',
+                                                     'url': 'https?://commons.wikimedia.org/w/api.php'}],
+                                          'nl': [{'ext': 'srt',
+                                                  'url': 'https?://commons.wikimedia.org/w/api.php'}],
+                                          'en': [{'ext': 'srt',
+                                                  'url': 're:https?://commons.wikimedia.org/w/api.php'}]}}
+    }
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        ext = determine_ext(url, None)
+        if ext is None:
+            raise ExtractorError('invalid video url', expected=True)
```

Comment:
I don't think this check is needed. But you want it, do:
```suggestion
        if not determine_ext(url, None):
            raise ExtractorError('The URL does not contain a video', expected=True)
```

Or even better, the valid_url can be improved to not match URLs without extension

Reply:
I'm using this video extension later in the code but I added your suggestion for Error message